lua/zk/util.lua

```diff
@@ -0,0 +1,36 @@
+local lsp = require("zk.lsp")
+local config = require("zk.config")
+
+local M = {}
+
+function M.setup_lsp_auto_attach()
+  --- NOTE: modified version of code in nvim-lspconfig
+  local trigger
+  local filetypes = config.options.lsp.auto_attach.filetypes
+  if filetypes then
+    trigger = "FileType " .. table.concat(filetypes, ",")
+  else
+    trigger = "BufReadPost *"
+  end
+  vim.api.nvim_command(string.format("autocmd %s lua require'zk.util'.lsp_buf_auto_add(0)", trigger))
+end
+
+function M.is_notebook_path(path)
+  -- check that we got a match on the .zk root directory
+  return require("lspconfig.util").root_pattern(".zk")(path) ~= nil
```

Comment:
I'm sure it'll be quite obvious to most users, but we'll want to make sure that we note `nvim-lspconfig` as one of the project dependencies in the README

Reply:
Agreed! :) It's already in the README though, you must've overlooked it.